tools/zipit.py

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env python
+"""
+  simple tool ZIP output logfile into single ZIP archive
+    ./zipit.py <metalog filename>
+"""
+
+import sys
+import os
+from zipfile import ZipFile
+
+PREFIX = 'osgar_'
+
+
+def zipit(metalog_filename, output_filename):
+    dir_name = os.path.dirname(metalog_filename)
+    with ZipFile(output_filename, 'w') as myzip:
+        myzip.write(metalog_filename, os.path.basename(metalog_filename))
+        camera_file = None
+        for line in open(metalog_filename):
+            if line.startswith('['):
+                continue  # initial arguments
+            assert ': ' in line, line
+            name = line.split()[1].strip()
+            assert name.startswith('logs/')
+            filename = name[5:]
+            print filename
+            myzip.write(os.path.join(dir_name, filename), filename)
+            if filename.startswith('gps_'):
+                filename = filename.replace('.log', '.nmea')
+                filename = filename.replace('gps_', 'gps')
+                print filename
+                myzip.write(os.path.join(dir_name, filename), filename)
+            elif filename.startswith('camera_'):
+                camera_file = os.path.join(dir_name, filename)
+        
+        if camera_file is not None:
+            for line in open(camera_file):
+                if not line.startswith('('):
+                    continue
+                name = eval(line)[0]
+                assert name.startswith('logs/')
```

Comment:
Shouldn't be all the paths in the files be relative to the directory where the metalog file is placed? I am wondering how all these "logs/" starting paths are resolved when replaying the zip file or when navpat.py is started from a different directory.

Just something to think about.